tensorrt_llm/serve/openai_disagg_server.py

```diff
@@ -149,7 +162,58 @@ async def openai_completion(self, req: CompletionRequest) -> Response:
             logging.info("Sending request to gen server: %s", gen_server)
 
             if not gen_req.stream:
-                gen_response = await self.send_request(gen_server, gen_req)
+                gen_response = await self.send_completion_request(gen_server, gen_req)
+                return gen_response
+            else:
+                # Return a streaming response that combines both context and generation responses
+                return StreamingResponse(
+                    self.merge_streaming_responses(ctx_response, gen_server, gen_req),
+                    media_type="text/event-stream"
+                )
+
+        except CppExecutorError as e:
+            # If internal executor error is raised, shutdown the server
+            logging.exception(e)
+            signal.raise_signal(signal.SIGINT)
+        except HTTPException as e:
+            raise e  # Re-raise HTTP exceptions properly
+        except Exception as e:
+            logging.exception(e)
+            raise HTTPException(status_code=500, detail=f"Internal server error {str(e)}")
+
+    async def openai_chat_completion(self, req: ChatCompletionRequest) -> Response:
```

Comment:
Looks like most of the code is copied from `openai_completion` except for call to `send_chat_request` instead of `send_completion_request`. Can we refactor to avoid duplicating so much code?

Reply:
Thanks, look much better now.